table/tables/tables.go

```diff
@@ -137,33 +159,35 @@ func (t *Table) Cols() []*table.Column {
 	if len(t.publicColumns) > 0 {
 		return t.publicColumns
 	}
-
-	t.publicColumns = make([]*table.Column, 0, len(t.Columns))
+	publicColumns := make([]*table.Column, len(t.Columns))
```

Comment:
just for the first time it will be false. When everytimes we load schemas from tikv,  `t.publicColumns = t.Cols()` is called.